src/io/index.js

```diff
@@ -0,0 +1,308 @@
+                        if (err) return reject(err);
+                        resolve(`Successfully removed ${path}.`);
+                    });
+                }
+            });
+        });
+    }
+
+    // Deletes a given Directory /A
+    async deleteDirectory(path) {
+        return new Promise((resolve, reject) => {
+            fs.rmdir(path, function(err) {
+                if (err) return reject(err);
+                resolve();
+            });
+        });
+    }
+
+    // Deletes a given File /A.txt
+    async deleteFile(path) {
+        return new Promise((resolve, reject) => {
+            fs.unlink(path, function(err) {
+                if (err) return reject(err);
+                resolve();
+            });
+        });
+    }
+
+    // Format filesystem
+    async clearFileSystem() {
+        return new Promise((resolve, reject) => {
+            formatFS(function(err) {
+                if (err) return reject(err);
+                resolve();
+            });
+        });
+    }
+
+    // Creates a given Path /A
+    async createPath(path) {
+        return new Promise((resolve, reject) => {
+            sh.mkdirp(path, function(err) {
+                if (err) return reject(err);
+                resolve();
+            });
+        });
+    }
+
+    // Retrieves a file
+    async getFile(path) {
+        // TODO: need to add promises to Filer
+        return new Promise((resolve, reject) => {
+            fs.stat(path, (err, stats) => {
+                if (err) {
+                    return reject(err);
+                } else {
+                    // If this is a dir, show a dir listing
+                    if (stats.isDirectory()) {
+                        // Todo: Better error handling needed.
+                        return reject('Path does not link to a File.');
+                    } else {
+                        fs.readFile(path, (err, contents) => {
+                            if (err) {
+                                return reject(err);
+                            }
+                            resolve({
+                                type: getMimeType(path),
+                                body: contents,
+                            });
+                        });
+                    }
+                }
+            });
+        });
+    }
+
+    // Gets file info
+    async getFileInfo(path) {
+        // TODO: need to add promises to Filer
+        return new Promise((resolve, reject) => {
+            fs.stat(path, (err, stats) => {
+                if (err) {
+                    return reject(err);
+                } else {
+                    // If this is a dir, show a dir listing
+                    if (stats.isDirectory()) {
+                        // Todo: Better error handling needed.
+                        return reject('Path does not link to a File.');
+                    } else {
+                        fs.readFile(path, 'utf8', (err, contents) => {
```

Comment:
I don't understand this.  If it's not a text file, this will return something unusable.  Just stat the file and give back the info.  If you want the contents, you need to read it, and probably you should just send them `/www/path` and it will automatically get the file for you with proper mime type.